### DISCUSSION TITLE
Hi.About spectype in ruleset.

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#769] [Q&A] [answered]
Hi, I'm using ruleset to meet my needs.
If I use ruletype: RuleTypes.Content Type, equivalent to I chose RuleTypes.Content Such a type, I can only use its related properties.

If I use spectype: conte ntSpecificationTypes.ContentRelatedInstances It's like I chose conte ntSpecificationTypes.ContentRelatedInstances Type, only its related definitions can be used.

Is my understanding correct?
If it is correct, I now find that the following writing is correct, although it does not take effect.**(According to the principle, it should report an error, because I used a property that does not belong to it)**

The three properties of **acceptablepolymerally**, **acceptableclassnames** and **acceptableschemaname** belong to context ntSpecificationTypes.SelectedNodeInstances , but does not belong to conte ntSpecificationTypes.ContentRelatedInstances I used it in this way, but didn't report it wrong. Although it doesn't work, is it my understanding?

        {
          specType: ContentSpecificationTypes.ContentRelatedInstances,
          relationshipPaths: [
            {
              relationship: {
                schemaName: "BisCore",
                className: "ModelContainsElements",
              },
              direction: RelationshipDirection.Forward,
              targetClass: {
                schemaName: "BisCore",
                className: "Element",
              },
            },
          ],
          acceptablePolymorphically: true,
          acceptableClassNames: [""],
          acceptableSchemaName: "bisCore",
        },

**Comment by grigasp** [comment, marked as answer]
Hi,

There are two ways to define presentation rules:
1. As a `.json` file. In this case error and type checking are done by your editor, assuming that you have specified the `$schema` attribute pointing to `@bentley/imodeljs-common/Ruleset.schema.json`.
2. As a TS/JS object in code. In this case error and type checking is done by TS type checking system.

Either way, if you have error / type checking disabled, there's nothing that prevents you from specifying excessive attributes to any rule - they're simply ignored. 

I see in your example you're defining the rules in TS. To enable type checking, specify the type of object you're creating:
```ts
import { Ruleset } from "@bentley/presentation-common";

const rules: Ruleset = {
  // ...
};
```

**Reply by shenshengyi** [reply]
Thank you very much. It helped me a lot.
